doc/kj.md

```diff
@@ -0,0 +1,84 @@
+---
+layout:	page
+title:	KJ
+---
+
+# KJ: A Performance and Safety Library for C++
+
+KJ provides a variety of types, templates, and macros to ease the writing of correct,
+high performance C++ code.
+
+## Primitives
+
+### Maybe (optional)
+
+`Maybe<T>` is a template for an optionally contained value, similar to (but predating)
+[`std::optional<T>`](https://en.cppreference.com/w/cpp/utility/optional). It has the advantage of
+ensuring that the empty case is handled, thus preventing null pointer dereferences and/or access to
+invalid objects at runtime.
+
+You can construct a `Maybe<T>` using either a value of type `T` or using `nullptr`:
+
+```c++
+#include <kj/common.h>
+
+auto number = kj::Maybe<double>(1.0);
+auto empty = kj::Maybe<double>(nullptr);
```

Comment:
This is of course valid, but I think more typical code looks like defining a `kj::Maybe<double> foo;` that is empty in advance, or as part of a class field and then later filling it in (or not!).